state/redis/redis.go

```diff
@@ -109,23 +128,45 @@ func (r *StateStore) Init(metadata state.Metadata) error {
 		return err
 	}
 	r.metadata = m
+	var redisClient *redis.Client
+
+	if r.metadata.failover {
+		opts := &redis.FailoverOptions{
+			MasterName:      r.metadata.sentinelMasterName,
+			SentinelAddrs:   []string{r.metadata.host},
+			DB:              defaultDB,
+			MaxRetries:      m.maxRetries,
+			MaxRetryBackoff: m.maxRetryBackoff,
+		}
 
-	opts := &redis.Options{
-		Addr:            m.host,
-		Password:        m.password,
-		DB:              defaultDB,
-		MaxRetries:      m.maxRetries,
-		MaxRetryBackoff: m.maxRetryBackoff,
-	}
+		/* #nosec */
+		if m.enableTLS {
+			opts.TLSConfig = &tls.Config{
+				InsecureSkipVerify: m.enableTLS,
+			}
+		}
 
-	/* #nosec */
-	if m.enableTLS {
-		opts.TLSConfig = &tls.Config{
-			InsecureSkipVerify: m.enableTLS,
+		redisClient = redis.NewFailoverClient(opts)
+	} else {
+		opts := &redis.Options{
+			Addr:            m.host,
+			Password:        m.password,
+			DB:              defaultDB,
+			MaxRetries:      m.maxRetries,
+			MaxRetryBackoff: m.maxRetryBackoff,
 		}
+
+		/* #nosec */
+		if m.enableTLS {
```

Comment:
Can we move this if statement to line 165 and remove the duplicate code?

Reply:
Move them to function now. Does this good?